### PR TITLE
feat: support github private repository

### LINKED ISSUE #96: Sync github private repo
Hello @pahud,

Thank you for creating such a great package.

Would you please guide me on how to sync the private repo?

Thanks,

### PR DESCRIPTION
**Pull request by pahud**
This PR allows you to sync a **private** github repository to the efs filesystem.

Fixes #96

### CODE REVIEW

**Comment by pahud** [comment]
Hi @semsphy  the PR is ready. Please let me know if the API design is relevant to this feature request.

@guan840912 can you help review this PR? Thanks.

**Comment by pahud** [comment]
Hi @stephenwiebe 

Any comments on this PR?

**Comment by semsphy** [comment]
> Hi @semsphy  the PR is ready. Please let me know if the API design is relevant to this feature request.
> 
> @guan840912 can you help review this PR? Thanks.

@pahud , it is 💯 relevant. Really appreciate.

**Comment by semsphy** [comment]
> Hi @stephenwiebe 
> 
> Any comments on this PR?

@pahud, to me I think we should remove the `token` placeholder and do the magic behind scene.

File: src/integ.default.ts [hunk 1]
```diff
@@ -48,6 +49,29 @@ export class IntegTesting {
...
+      syncSource: SyncSource.github({
+        vpc: vpc,
+        repository: 'https://{token}@github.com/pahud/private-repo.git',
+        secret: {
+          id: 'github',
+          key: 'oauth_token',
+        },
+      }),
+    });
+
```
Comment (guan840912) [inline src/integ.default.ts:74]: Maybe can add new props like "repoType", to define public or private repo,and then to reassemble repo uri for private or public!?

**Comment by pahud** [comment]
> > Hi @stephenwiebe
> > Any comments on this PR?
> 
> @pahud, to me I think we should remove the `token` placeholder and do the magic behind scene.

removed. 

Please check it out again.

**Comment by semsphy** [comment]
> > > Hi @stephenwiebe
> > > Any comments on this PR?
> > 
> > @pahud, to me I think we should remove the `token` placeholder and do the magic behind scene.
> 
> removed. 
> 
> Please check it out again.

Cool man, that's the magic ✨

**Review by guan840912 — APPROVED** [review]

File: src/integ.default.ts [hunk 2]
```diff
@@ -48,6 +49,29 @@ export class IntegTesting {
...
+        ownerUid: '1001',
+        permissions: '0755',
+      },
+      posixUser: {
+        uid: '1001',
+        gid: '1001',
+      },
+      syncSource: SyncSource.github({
+        vpc: vpc,
+        repository: 'https://{token}@github.com/pahud/private-repo.git',
```
[the remark this answers is not in the document]
Comment (pahud) [reply src/integ.default.ts:67]: thanks! will remove it.

File: src/integ.default.ts (hunk 1 above)
Comment (pahud) [reply src/integ.default.ts:74]: Per our offline discussion, we are going to leave it as is to keep it simple and clean.

**Pull request merged**